File: python/customer_base.py
class CustomerBase:

    def __init__(self):
        self.customers = []

    def add(self, customer):
        self.customers.append(customer)

    def findByLastName(self, lastName):
        result = []
        for customer in self.customers:
            if customer.lastName == lastName:
                result.append(customer)
        return result

    def findByFirstAndLastName(self, firstName, lastName):
        result = []
        for customer in self.customers:
            if customer.firstName == firstName and customer.lastName == lastName:
                result.append(customer)
        return result

    def findByCreditGreaterThan(self, credit):
        result = []
        for customer in self.customers:
            if customer.credit > credit:
                result.append(customer)
        return result

File: csharp/CustomerBase.cs
using System;
using System.Collections.Generic;

namespace CustomerBase
{
	public class CustomerBase
	{
		private List<Customer> customers = new List<Customer>();

		public void Add(Customer customer)
		{
			customers.Add(customer);
		}

		public List<Customer> FindByLastName(string lastName)
		{
			List<Customer> result = new List<Customer>();
			foreach (Customer customer in customers)
				if (customer.LastName == lastName)
					result.Add(customer);
			return result;
		}

		public List<Customer> FindByFirstAndLastName(string firstName, string lastName)
		{
			List<Customer> result = new List<Customer>();
			foreach (Customer customer in customers)
				if (customer.FirstName == firstName && customer.LastName == lastName )
					result.Add(customer);
			return result;
		}

		public List<Customer> FindByCreditGreaterThan(int credit)
		{
			List<Customer> result = new List<Customer>();
			foreach (Customer customer in customers)
				if (customer.Credit > credit)
					result.Add(customer);
			return result;
		}
	}
}
